arukas/resource_arukas_container.go

```diff
@@ -18,46 +18,52 @@ func resourceArukasContainer() *schema.Resource {
 		Importer: &schema.ResourceImporter{
 			State: schema.ImportStatePassthrough,
 		},
+		SchemaVersion: 1,
 		Schema: map[string]*schema.Schema{
-			"name": &schema.Schema{
+			"name": {
 				Type:     schema.TypeString,
 				Required: true,
 				ForceNew: true,
 			},
-			"image": &schema.Schema{
+			"image": {
 				Type:     schema.TypeString,
 				Required: true,
 			},
-			"instances": &schema.Schema{
+			"instances": {
 				Type:         schema.TypeInt,
 				Optional:     true,
 				Default:      1,
 				ValidateFunc: validateIntegerInRange(1, 10),
 			},
-			"memory": &schema.Schema{
-				Type:         schema.TypeInt,
+			"memory": {
+				Type:     schema.TypeInt,
+				Optional: true,
+				Removed:  "Use `plan` instead. This attribute will be removed in a future version",
+			},
+			"plan": {
+				Type:         schema.TypeString,
 				Optional:     true,
-				Default:      256,
-				ValidateFunc: validateIntInWord([]string{"256", "512"}),
+				Default:      arukas.PlanFree,
+				ValidateFunc: validateStringInWord(arukas.ValidPlans),
```

Comment:
this can just be [`validation.StringInSlice`](https://github.com/hashicorp/terraform/blob/c0793c84fdbea8c4f3f7597cad57bcca38b21ebf/helper/validation/validation.go#L76)